### PR TITLE
API for EDP Metrics

### PR DESCRIPTION
**Pull request by karthikeyan-cere**

### CODE REVIEW

File: api/index.js [hunk 1]
```diff
@@ -215,6 +234,71 @@ app.get('/api/v1/batsignal/council-events', async (req, res) => {
...
+/**
+ * Add EDP data to database
+ * @param {key, values} key and values
+ */
+app.post('/api/v1/edp', validateToken, async (req, res) => {
+  try {
+    const { key, values } = req.body;
+    const client = await getClient();
+    const query = `
+    INSERT INTO edp_metric (
```
Comment (AndreiNavoichyk) [inline api/index.js]: Apply the same formatting for the query as above.

File: api/index.js [hunk 2]
```diff
@@ -215,6 +234,71 @@ app.get('/api/v1/batsignal/council-events', async (req, res) => {
...
+      value
+    )
+    VALUES (
+      $1,
+      $2
+    )
+    ON CONFLICT (key) DO UPDATE
+    SET value = EXCLUDED.value
+    WHERE EXCLUDED.key = $1
+    `;
```
Comment (AndreiNavoichyk) [inline api/index.js]: Add semicolon inside query

File: api/index.js [hunk 3]
```diff
@@ -215,6 +234,71 @@ app.get('/api/v1/batsignal/council-events', async (req, res) => {
...
+    const dbres = await client.query(query, [key, JSON.stringify(values)]);
+    res.status(200).json({
+      status: true,
+      data: dbres,
```
Comment (AndreiNavoichyk) [inline api/index.js]: We can not return DB response to public. Just status is enough

**Review by AndreiNavoichyk — COMMENTED** [review]
Add changelog.

File: api/index.js (hunk 3 above)
Comment (karthikeyan-cere) [reply api/index.js]: Updated response

File: api/index.js (hunk 1 above)
Comment (karthikeyan-cere) [reply api/index.js]: added

File: api/index.js [hunk 4]
```diff
@@ -215,6 +234,69 @@ app.get('/api/v1/batsignal/council-events', async (req, res) => {
...
+ * Get EDP data from database
+ * @param key key
+ */
+app.get('/api/v1/edp/:key', validateToken, async (req, res) => {
+  try {
+    const { key } = req.params;
+    const client = await getClient();
+    const query = `SELECT value FROM edp WHERE key = $1;`;
+    const dbres = await client.query(query, [key]);
+    res.status(200).json({
```
Comment (AndreiNavoichyk) [inline api/index.js:284]: What will the api return in case if nothing in the DB for this key?

File: api/index.js [hunk 5]
```diff
@@ -215,6 +234,69 @@ app.get('/api/v1/batsignal/council-events', async (req, res) => {
...
+/**
+ * Add EDP data to database
+ * @param {key, value} key and values
+ */
+app.post('/api/v1/edp', validateToken, async (req, res) => {
+  try {
+    const { key, value } = req.body;
+    const client = await getClient();
+    const query = `
+    INSERT INTO edp (
```
Comment (AndreiNavoichyk) [inline api/index.js]: Fix formatting. Take a look to the queries above.

File: api/index.js (hunk 4 above)
Comment (karthikeyan-cere) [reply api/index.js:284]: It will be empty

Comment (AndreiNavoichyk) [reply api/index.js:284]: But we should return 404.

Comment (karthikeyan-cere) [reply api/index.js:284]: added

File: api/postman/CereStats.postman_collection.json [hunk 6]
```diff
@@ -0,0 +1,73 @@
+{
+	"info": {
+		"_postman_id": "b5772ee7-2479-4ffb-b10e-9f95134778b0",
+		"name": "CereStats",
```
Comment (AndreiNavoichyk) [inline api/postman/CereStats.postman_collection.json]: Let's use 1 word `Cerestats`

File: api/postman/CereStats.postman_collection.json [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+	},
+	"item": [
+		{
+			"name": "Add value",
+			"request": {
+				"method": "POST",
+				"header": [
+					{
+						"key": "X-API-KEY",
+						"value": "jP9J5rUCKg7sWJ9y75eT",
```
Comment (AndreiNavoichyk) [inline api/postman/CereStats.postman_collection.json]: Use {x-api-key-goes-here}

File: api/postman/CereStats.postman_collection.json [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+			"response": []
+		},
+		{
+			"name": "Get data",
+			"request": {
+				"method": "GET",
+				"header": [
+					{
+						"key": "X-API-KEY",
+						"value": "jP9J5rUCKg7sWJ9y75eT",
```
Comment (AndreiNavoichyk) [inline api/postman/CereStats.postman_collection.json]: Use {x-api-key-goes-here}

File: api/postman/CereStats.postman_collection.json [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+					"mode": "raw",
+					"raw": "{\n    \"key\" : \"test1\",\n    \"value\": {\n        \"graduates-number\": \"10\"\n    }\n}",
+					"options": {
+						"raw": {
+							"language": "json"
+						}
+					}
+				},
+				"url": {
+					"raw": "http://localhost:8000/api/v1/edp",
```
Comment (AndreiNavoichyk) [inline api/postman/CereStats.postman_collection.json]: Use Cerestats PROD host

File: api/postman/CereStats.postman_collection.json [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+				"method": "GET",
+				"header": [
+					{
+						"key": "X-API-KEY",
+						"value": "jP9J5rUCKg7sWJ9y75eT",
+						"type": "text"
+					}
+				],
+				"url": {
+					"raw": "http://localhost:8000/api/v1/edp/test1",
```
Comment (AndreiNavoichyk) [inline api/postman/CereStats.postman_collection.json]: Use Cerestats PROD host

File: api/postman/CereStats.postman_collection.json [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+					"protocol": "http",
+					"host": [
+						"localhost"
+					],
+					"port": "8000",
+					"path": [
+						"api",
+						"v1",
+						"edp",
+						"test1"
```
Comment (AndreiNavoichyk) [inline api/postman/CereStats.postman_collection.json]: Remove extra data

Comment (karthikeyan-cere) [reply api/postman/CereStats.postman_collection.json]: updated

File: api/postman/CereStats.postman_collection.json (hunk 10 above)
Comment (karthikeyan-cere) [reply api/postman/CereStats.postman_collection.json]: updated

File: api/postman/CereStats.postman_collection.json (hunk 9 above)
Comment (karthikeyan-cere) [reply api/postman/CereStats.postman_collection.json]: updated

File: api/postman/CereStats.postman_collection.json (hunk 8 above)
Comment (karthikeyan-cere) [reply api/postman/CereStats.postman_collection.json]: updated

File: api/postman/CereStats.postman_collection.json (hunk 7 above)
Comment (karthikeyan-cere) [reply api/postman/CereStats.postman_collection.json]: updated

File: api/postman/CereStats.postman_collection.json (hunk 6 above)
Comment (karthikeyan-cere) [reply api/postman/CereStats.postman_collection.json]: updated

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**